### PR TITLE
Update the Azure Batch SDK to support REST API version 2018-08-01.7.0

### PR DESCRIPTION
**Pull request by matthchr**
## Description
Update the Azure Batch SDK to support REST API version 2018-08-01.7.0

**This is for a 5.0.0-preview release of the AzureRM.Batch module**. There are some (small) breaking changes.

## Checklist

- [x] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log)
- [ ] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr)
    - [x] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/help-generation.md)

### CODE REVIEW

**Comment by matthchr** [comment]
@cormacpayne - As mentioned over IM, my main uncertainty is if I got the changelog/version stuff right, so that's definitely worth a review.

Note that you can review by commit with might help avoid the huge amount of regenerated test files

**Comment by matthchr** [comment]
Oh, we should announce our breaking changes in the breaking changes file too, right?

File: src/ResourceManager/AzureBatch/Commands.Batch/Commands.Batch.csproj [hunk 1]
```diff
@@ -42,8 +42,8 @@
     <Prefer32Bit>false</Prefer32Bit>
   </PropertyGroup>
   <ItemGroup>
-    <Reference Include="Microsoft.Azure.Batch, Version=8.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
-      <HintPath>..\..\..\packages\Azure.Batch.8.1.1\lib\net452\Microsoft.Azure.Batch.dll</HintPath>
+    <Reference Include="Microsoft.Azure.Batch, Version=9.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
+      <HintPath>..\..\..\packages\Microsoft.Azure.Batch.9.0.0\lib\net452\Microsoft.Azure.Batch.dll</HintPath>
```
Comment (cormacpayne) [inline src/ResourceManager/AzureBatch/Commands.Batch/Commands.Batch.csproj]: @matthchr you'll want to make sure the version of your SDK in [`Commands.Batch.Netcore.csproj`](https://github.com/Azure/azure-powershell/blob/preview/src/ResourceManager/AzureBatch/Commands.Batch/Commands.Batch.Netcore.csproj#L35)

File: src/ResourceManager/AzureBatch/Commands.Batch/Az.Batch.psd1 [hunk 2]
```diff
@@ -12,7 +12,7 @@
 # RootModule = ''
 
 # Version number of this module.
-ModuleVersion = '0.2.2'
+ModuleVersion = '0.3.0'
```
Comment (cormacpayne) [inline src/ResourceManager/AzureBatch/Commands.Batch/Az.Batch.psd1]: @matthchr you can change this to `0.2.3` -- we plan on doing minor version bumps for Cloud Shell releases every two weeks (until we GA and release as `1.0.0`), so these one-off "preview" releases should do patch version bumps

File: src/ResourceManager/AzureBatch/Commands.Batch/Az.Batch.psd1 [hunk 3]
```diff
@@ -150,7 +150,7 @@ PrivateData = @{
         ReleaseNotes = 'Initial Release with PowerShell and PowerShell Core Support'
 
         # Prerelease string of this module
-        # Prerelease = ''
+        Prerelease = 'preview'
```
Comment (cormacpayne) [inline src/ResourceManager/AzureBatch/Commands.Batch/Az.Batch.psd1]: @matthchr you can revert the change here -- for `Az`, we're following the semantic versioning pattern that `0.x.x` is preview (or unstable), so we won't need to do `0.x.x-preview`

File: src/ResourceManager/AzureBatch/Commands.Batch/AzureRM.Batch.psd1 [hunk 4]
```diff
@@ -147,11 +147,12 @@ PrivateData = @{
         # IconUri = ''
 
         # ReleaseNotes of this module
-        ReleaseNotes = '* Fixed issue with default resource groups not being set.
-* Updated common runtime assemblies'
+        ReleaseNotes = '* Added the ability to see what version of the Azure Batch Node Agent is running on each of the VMs in a pool, via the new `NodeAgentInformation` property on `PSComputeNode`.
```
Comment (cormacpayne) [inline src/ResourceManager/AzureBatch/Commands.Batch/AzureRM.Batch.psd1]: @matthchr the module version at the top of this file should be `5.0.0`

**Comment by cormacpayne** [comment]
@matthchr as for announcing the breaking changes, you can create a pull request against the `preview` branch with the deprecation messages so users will get them for the next release. You will also need to log the breaking change in the [`upcoming-breaking-changes.md`](https://github.com/Azure/azure-powershell/blob/preview/src/ResourceManager/AzureBatch/documentation/upcoming-breaking-changes.md) file.

File: src/ResourceManager/AzureBatch/Commands.Batch/Az.Batch.psd1 (hunk 2 above)
Comment (matthchr) [reply src/ResourceManager/AzureBatch/Commands.Batch/Az.Batch.psd1]: Okay cool, will fix.

File: src/ResourceManager/AzureBatch/Commands.Batch/Commands.Batch.csproj (hunk 1 above)
Comment (matthchr) [reply src/ResourceManager/AzureBatch/Commands.Batch/Commands.Batch.csproj]: Done -- do I need to do any extra testing around this...?

**Comment by matthchr** [comment]
See #7426 for the breaking change announcements.

**Comment by matthchr** [comment]
This now also fixes #7005

File: src/ResourceManager/AzureBatch/Commands.Batch/Commands.Batch.csproj (hunk 1 above)
Comment (cormacpayne) [reply src/ResourceManager/AzureBatch/Commands.Batch/Commands.Batch.csproj]: @matthchr we're in the process of converting all of the desktop test projects to .NET Standard, so we're looking to run all tests on Travis soon. As long as the coverage is in the desktop version, we should feel fairly confident it works on .NET Standard as well.
